### ISSUE TITLE
ArrayShapes

### ISSUE THREAD

**Issue by puggan** [spatie/typescript-transformer#68]
phpdocs with arrayshapes fails.

for exemple: `/** @var array{a: int[], b: string[], c: float} $config **/`

first it fails at the regexp:
https://github.com/spatie/typescript-transformer/blob/37eda5f0f72cce8008b8b4cf0c802e1d35428812/src/TypeReflectors/PropertyTypeReflector.php#L27

then it fails at transpiling ([phpDocumentor\Reflection\PseudoTypes\ArrayShape](https://github.com/phpDocumentor/TypeResolver/blob/1.x/src/PseudoTypes/ArrayShape.php))
https://github.com/spatie/typescript-transformer/blob/af3aca5716adede7dd3300e12447ec08e7131655/src/Actions/TranspileTypeToTypeScriptAction.php#L60

it's partly solved with:
https://github.com/spatie/typescript-transformer/blob/37eda5f0f72cce8008b8b4cf0c802e1d35428812/src/Actions/TranspileTypeToTypeScriptAction.php#L61

Senario 1

Input: `/** @var array<string, array{a: int[], b: string[], c: float}> $config **/`

Expected: `{[key: string]: {a: number[], b: string[], c: number}}`

Current: RuntimeException `Unexpected token "", expected '>' at offset 19 on line 1`,
due to the regexp only catching ` @var array<string, array`

**Comment by rubenvanassche** [comment]
Yeah, this is something we're going to handle in v3 which is next big thing I'm working on, it is scomplete rewrite of TS transformer using PHPStan for those things instead of phpDocumentor.

**Comment by puggan** [comment]
Could you update the thrown exception to include what class it failed to parse?

as a workaround i put an `echo $class->name, PHP_EOL;`
in \Spatie\TypeScriptTransformer\Actions\ResolveTypesCollectionAction::resolveTransformedType
to find out what fail I needed to add an `#[LiteralTypeScriptType()]` in.
